service/matching/task_queue_partition_manager.go

```diff
@@ -1075,3 +1085,31 @@ func (pm *taskQueuePartitionManagerImpl) getPerTypeUserData() (*persistencespb.T
 	perType := userData.GetData().GetPerType()[int32(pm.Partition().TaskType())]
 	return perType, userDataChanged, nil
 }
+
+// makePollerScalingDecision makes a decision on whether to scale pollers up or down based on the current state of the
+// task queue and the task about to be returned. Does not modify inputs.
+func (pm *taskQueuePartitionManagerImpl) makePollerScalingDecision(
+	stats *taskqueuepb.TaskQueueStats, task *internalTask, pollStartTime time.Time,
+) *taskqueuepb.PollerScalingDecision {
+	pd := &taskqueuepb.PollerScalingDecision{}
+	pollWaitTime := pm.engine.timeSource.Since(pollStartTime)
+	if stats.ApproximateBacklogCount > 0 && stats.ApproximateBacklogAge.AsDuration() > pm.config.PollerScalingBacklogAgeScaleUp() {
+		// Always increase when there is a backlog, even if we're a partition. It's also important to increase for
+		// sticky queues.
+		pd.PollRequestDeltaSuggestion = 1
+	} else if !pm.partition.IsRoot() {
+		// Non-root partitions don't have an appropriate view of the data to make decisions beyond backlog.
+		pd = nil
+	} else if task.source == serverenumspb.TASK_SOURCE_HISTORY &&
```

Comment:
Why do we check source here? Maybe a comment? The existing comment just says "if any poll matched after sitting idle", which doesn't have anything to do with the source of the task

Reply:
Well, I wanted to see that it was a sync match and when we discussed it you mentioned this would be the preferred way to do that. Maybe that doesn't really matter though, I suppose any reason we waited is just as valid.

Reply:
Yeah, my general feeling is the more we ignore stuff like the source of a task, the easier it is to understand overall